### DISCUSSION TITLE
Question about indexing a tensor - X[1,2,3] vs X[1][2][3]

### DISCUSSION THREAD

**Discussion by pbsepto82** [mrdbourke/pytorch-deep-learning#548] [General]
Hello!

I was trying out indexing a tensor and to my surprise, I found indexing returns different results depending on whether I use a single square bracket (with comma separated indices) OR multiple square brackets (with each index having it's own bracket). Here's my code:-

`import torch

tensorA = torch.arange(1,19).reshape(2,3,3)
print(tensorA)
print(tensorA[0][:][2])

print("\n")

x = torch.arange(1,19).reshape(2,3,3)
print(x)
print(x[0,:,2])`

This is what the output is:-

tensor([[[ 1,  2,  3],
         [ 4,  5,  6],
         [ 7,  8,  9]],

        [[10, 11, 12],
         [13, 14, 15],
         [16, 17, 18]]])
tensor([7, 8, 9])

tensor([[[ 1,  2,  3],
         [ 4,  5,  6],
         [ 7,  8,  9]],

        [[10, 11, 12],
         [13, 14, 15],
         [16, 17, 18]]])
tensor([3, 6, 9])

My intention is to get the second outcome but I'm not able to understand why the first one is returning tensor([7, 8, 9])

I'm very new to this, would really appreciate if anyone can guide. Thanks in advance!

**Comment by Perian-Yan** [comment]
Hi, 
I think the problem comes from slicing [:], which is different when using a single bracket and multiple brackets. You can print tensorA step by step and see what's going on: 
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/100858995/41063e63-331c-430e-80f8-10de29817346)

For indexing, I think they are the same.
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/100858995/810245f2-42e6-4bce-9817-fb575268f45d)

**Reply by pbsepto82** [reply]
Thank you!
I see, it's little misleading though I'm specifying index 2 for 3rd dimension to get the 2nd values obviously, but the result is not in accordance with what it looks like with [0][:][2]. Would I be right to say that we should always use single bracket when slicing to avoid any confusion?

**Reply by Perian-Yan** [reply]
Yes, to get, e.g., tensor([3,6,9]), we can only use single bracket slicing.
